src/timeUtils.ts

```diff
@@ -861,85 +869,100 @@ export function getPGIntervalUnixEpochTime(startTimeMs: number, endTimeMs: numbe
 }
 
 /**
- * Get a Unix epoch time in milliseconds given a day-of-year (DOY) timestamp string.
+ * Get a Unix epoch time in milliseconds given a ISO Ordinal and ISO 8601 UTC timestamp string.
  *
  * @function getUnixEpochTime
- * @param {string} doyTimestamp - The DOY timestamp string in the format "YYYY-DOYTHH:MM:SS[.mmm]".
+ * @param {string} isoOrUtcTime - The ISO Ordinal and ISO 8601 UTC timestamp string in the format "YYYY-DOYTHH:MM:SS[.mmm]" or "YYYY-MM-DDTHH:mm:ssZ".
  * @returns {number} The Unix epoch time in milliseconds. Returns 0 if the timestamp is invalid.
  *
  * @example
  * getUnixEpochTime('2019-365T08:00:00.000') // returns 1577779200000
- *
+ * getUnixEpochTime('2023-12-27T10:30:00Z') // returns 1577779200000
+ * 
  * @note Inverse of getDoyTime.
  */
-export function getUnixEpochTime(doyTimestamp: string): number {
-  const re = /(\d{4})-(\d{3})T(\d{2}):(\d{2}):(\d{2})\.?(\d{3})?/;
-  const match = re.exec(doyTimestamp);
-
-  if (match) {
-    const [, year, doy, hours, mins, secs, msecs = '0'] = match;
-    return Date.UTC(+year, 0, +doy, +hours, +mins, +secs, +msecs);
+export function getUnixEpochTime(isoOrUtcTime: string): number {
+  let match = ISO_ORDINAL_TIME_REGEX.exec(isoOrUtcTime);
+  if (match && match.groups) {
+    const { year, doy, hr, mins, secs, ms = '0'} = match.groups;
+    return Date.UTC(+year, 0, +doy, +hr, +mins, +secs, +ms);
+    Date.UTC(0  ,0,0,0,0,0,36000)
```

Comment:
does this line do anything after the return?

Reply:
opps nope not sure what I was doing there :P